zkevm-circuits/src/evm_circuit/execution/extcodecopy.rs

```diff
@@ -50,17 +50,17 @@ impl<F: Field> ExecutionGadget<F> for ExtcodecopyGadget<F> {
         let external_address =
             from_bytes::expr(&external_address_word.cells[..N_BYTES_ACCOUNT_ADDRESS]);
 
-        let memory_offset = cb.query_cell_phase2();
-        let data_offset = cb.query_word_rlc();
+        let code_size = cb.query_cell();
+
         let memory_length = cb.query_word_rlc();
+        let memory_offset = cb.query_cell_phase2();
+        let code_offset = WordByteCapGadget::construct(cb, code_size.expr());
```

Comment:
We're using `WordByteCapGadget` for `code_offset`, but I don't see any place where we check `code_offset.overflow` or `code_offset.within_range`.  So the internal `IsZero` gadget seems to be unused?

Reply:
The `within_range` is invoked in [WordByteCapGadget::construct](https://github.com/privacy-scaling-explorations/zkevm-circuits/blob/4419f2bdbaca12900df3d577556e28777b4823a3/zkevm-circuits/src/evm_circuit/util/common_gadget.rs#L1052) to get valid value in range, then compare with `cap` (code-length for `CODECOPY`). It is used to get the correct memory start.

Since the code-offset is set to `u64::MAX` if overflow in [geth opCodeCopy](https://github.com/ethereum/go-ethereum/blob/master/core/vm/instructions.go#L363). And the memory offset is reset to code-length (if code-offset is greater than code-length) in [getData](https://github.com/ethereum/go-ethereum/blob/66c0c4e5174a0b231727f27c7ad48624189ecc62/core/vm/common.go#L57).

The `lt_cap` is used to set memory start in [this code](https://github.com/privacy-scaling-explorations/zkevm-circuits/blob/a3226b4d3c55f61a7cd9ef1258f6b7fc8d5c9d41/zkevm-circuits/src/evm_circuit/execution/codecopy.rs#L85).

BTW, the code-length could not be Uint64 overflow (otherwise raise an error of `ErrGasUintOverflow`).